bindings/python/py_src/tokenizers/__init__.pyi

```diff
@@ -26,6 +27,232 @@ PreTokenizedEncodeInput = Union[
+
+            offset_type: OffsetType:
+                Whether the returned splits should have offsets expressed in bytes or chars.
+                When slicing an str, we usually want to use chars, which is the default value.
+                Now in some cases it might be interesting to get these offsets expressed in bytes,
+                so it is possible to change this here.
+
+        Returns
+            A list of splits
+        """
+        pass
+
+class NormalizedString:
+    """ NormalizedString
+
+    A NormalizedString takes care of modifying an "original" string, to obtain a "normalized" one.
+    While making all the requested modifications, it keeps track of the alignment information
+    between the two versions of the string.
+    """
+
+    def __new__(sequence: str) -> NormalizedString:
+        """ Instantiate a new NormalizedString using the given str
+
+        Args:
+            sequence: str:
+                The string sequence used to initialize this NormalizedString
+        """
+        pass
+    @property
+    def normalized(self) -> str:
+        """ The normalized part of the string """
+        pass
+    @property
+    def original(self) -> str:
+        """ The original part of the string """
+        pass
+    def nfd(self):
+        """ Runs the NFD normalization """
+        pass
+    def nfkd(self):
+        """ Runs the NFKD normalization """
+        pass
+    def nfc(self):
+        """ Runs the NFC normalization """
+        pass
+    def nfkc(self):
+        """ Runs the NFKC normalization """
+        pass
+    def lowercase(self):
+        """ Lowercase the string """
+        pass
+    def uppercase(self):
+        """ Uppercase the string """
+        pass
+    def prepend(self, s: str):
+        """ Prepend the given sequence to the string """
+        pass
+    def append(self, s: str):
+        """ Append the given sequence to the string """
+        pass
+    def lstrip(self):
+        """ Strip the left of the string """
+        pass
+    def rstrip(self):
+        """ Strip the right of the string """
+        pass
+    def strip(self):
+        """ Strip both ends of the string """
+        pass
+    def clear(self):
+        """ Clear the string """
+        pass
+    def slice(self, range: Range) -> Optional[NormalizedString]:
+        """ Slice the string using the given range """
+        pass
+    def filter(self, func: Callable[[str], bool]):
+        """ Filter each character of the string using the given func """
+        pass
+    def for_each(self, func: Callable[[str], None]):
+        """ Calls the given function for each character of the string """
+        pass
+    def map(self, func: Callable[[str], str]):
+        """ Calls the given function for each character of the string
+
+        Replaces each character of the string using the returned value. Each
+        returned value **must** be a str of length 1 (ie a character).
+        """
+        pass
```

Comment:
I think that's way too many primitives IMHO.
People should not use our internal API this way.

`replace` seems ok, could it be general enough that it can call `transform` automatically ? or maybe seems like `transform` is the way to go, no ?

Reply:
This is not internal API but public API. `transform` is too hard to be used on its own, and each method here make use of it for simplicity.